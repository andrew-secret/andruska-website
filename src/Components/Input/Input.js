import React from 'react';
import classnames from 'classnames';
import styles from './Input.module.sass';

const Input = props => {
  const inputClasses = classnames(styles.input, props.className);
  return (
    <div className={styles.wrapper}>
      <label className={styles.label} htmlFor={props.type}>
        {props.label}
      </label>
      {props.type === 'text' ? (
        <textarea className={styles.textarea} />
      ) : (
        <input
          className={inputClasses}
          type={props.type}
          id={props.type}
          required={props.required}
        />
      )}
    </div>
  );
};

export default Input;

Input.defaultProps = {
  type: 'name',
  label: 'This is a label',
  required: false,
};
