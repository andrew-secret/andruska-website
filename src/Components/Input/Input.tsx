import * as React from 'react';
import * as classnames from 'classnames';

const styles = require('./Input.module.sass');

interface Props {
  className?: string;
  label: string;
  type: 'text' | 'email' | 'name';
  required?: boolean;
}

export const Input: React.SFC<Props> = props => {
  const inputClasses = classnames(styles.input, props.className);
  return (
    <div className={styles.wrapper}>
      <label className={styles.label} htmlFor={props.type}>
        {props.label}
      </label>
      {props.type === 'text' ? (
        <textarea className={styles.textarea} />
      ) : (
        <input
          className={inputClasses}
          type={props.type}
          id={props.type}
          required={props.required}
        />
      )}
    </div>
  );
};
