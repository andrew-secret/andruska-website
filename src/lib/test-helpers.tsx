import * as shallowRenderer from 'react-test-renderer/shallow';
import { shallow, configure, ShallowWrapper } from 'enzyme';
import * as deepRenderer from 'react-test-renderer';
import * as Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

export type TestComponent = React.ReactElement<any>;

export const noOpFn = (): any => void 0;

// Renders component shallow with enzyme and lets you make assertions
export function renderComponentShallow(
  component: TestComponent
): ShallowWrapper {
  return shallow(component);
}

// Renders component shallow with react-test-renderer for snapshot tests
export function renderComponentShallowForSnapshot(component: TestComponent) {
  return shallowRenderer.createRenderer().render(component);
}

// Renders component deep without providing application environment
// Gives you access to full api of test renderer
export function renderComponent(component: TestComponent) {
  return deepRenderer.create(component);
}
