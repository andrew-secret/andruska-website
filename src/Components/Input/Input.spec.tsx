import * as React from 'react';
import { renderComponentShallowForSnapshot } from '../../lib/test-helpers';
import { Input } from './Input';

describe('Input', () => {
  it('Input name renders correctly', () => {
    const tree = renderComponentShallowForSnapshot(
      <Input label="some label" type="name" />
    );
    expect(tree).toMatchSnapshot();
  });
  it('Input text renders correctly', () => {
    const tree = renderComponentShallowForSnapshot(
      <Input label="some label" type="text" />
    );
    expect(tree).toMatchSnapshot();
  });
  it('Input email renders correctly', () => {
    const tree = renderComponentShallowForSnapshot(
      <Input label="some label" type="text" />
    );
    expect(tree).toMatchSnapshot();
  });
  it('Input is required renders correctly', () => {
    const tree = renderComponentShallowForSnapshot(
      <Input label="some label" type="text" required />
    );
    expect(tree).toMatchSnapshot();
  });
  it('Input with className renders correctly', () => {
    const tree = renderComponentShallowForSnapshot(
      <Input label="some label" type="text" className="someClass" />
    );
    expect(tree).toMatchSnapshot();
  });
});
