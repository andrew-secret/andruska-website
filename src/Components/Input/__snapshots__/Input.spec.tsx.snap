// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Input email renders correctly 1`] = `
<div
  className={undefined}
>
  <label
    className={undefined}
    htmlFor="text"
  >
    some label
  </label>
  <textarea
    className={undefined}
  />
</div>
`;

exports[`Input Input is required renders correctly 1`] = `
<div
  className={undefined}
>
  <label
    className={undefined}
    htmlFor="text"
  >
    some label
  </label>
  <textarea
    className={undefined}
  />
</div>
`;

exports[`Input Input name renders correctly 1`] = `
<div
  className={undefined}
>
  <label
    className={undefined}
    htmlFor="name"
  >
    some label
  </label>
  <input
    className=""
    id="name"
    required={undefined}
    type="name"
  />
</div>
`;

exports[`Input Input text renders correctly 1`] = `
<div
  className={undefined}
>
  <label
    className={undefined}
    htmlFor="text"
  >
    some label
  </label>
  <textarea
    className={undefined}
  />
</div>
`;

exports[`Input Input with className renders correctly 1`] = `
<div
  className={undefined}
>
  <label
    className={undefined}
    htmlFor="text"
  >
    some label
  </label>
  <textarea
    className={undefined}
  />
</div>
`;
